### PR TITLE
auto-editor: init at 25.3.0

### LINKED ISSUE #317015: Package request: auto-editor
Auto-Editor is a command line application for automatically editing video and audio by analyzing a variety of methods, most notably audio loudness.

**Metadata**

* homepage URL: https://auto-editor.com/
* source URL: https://github.com/WyattBlue/auto-editor
* license: The Unlicense
* platforms: linux (I'd guess also darwin and windows as it's all python)

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by TomaSajt**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/issues/317015

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/au/auto-editor/package.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    (substituteAll {
+      src = ./set-exe-paths.patch;
+      ffmpeg_path = lib.getExe ffmpeg;
+      yt_dlp_path = lib.getExe yt-dlp;
+    })
+  ];
+
+  postPatch = ''
+    # pyav is a fork of av, but has since been un-forked
+    substituteInPlace pyproject.toml \
```
Comment (Moraxyc) [inline pkgs/by-name/au/auto-editor/package.nix:40]: We can use `pythonRelaxDepsHook` in this case.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonrelaxdepshook-using-pythonrelaxdepshook

File: pkgs/by-name/au/auto-editor/package.nix [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    repo = "auto-editor";
+    rev = "refs/tags/${version}";
+    hash = "sha256-dXiNU9wPubmlZJesxvQYGvv1ga68wt/rsP6L2sWtYkE=";
+  };
+
+  patches = [
+    (substituteAll {
+      src = ./set-exe-paths.patch;
+      ffmpeg_path = lib.getExe ffmpeg;
+      yt_dlp_path = lib.getExe yt-dlp;
```
Comment (dotlambda) [inline pkgs/by-name/au/auto-editor/package.nix:34]:
```suggestion
      ffmpeg = lib.getExe ffmpeg;
      yt_dlp = lib.getExe yt-dlp;
```

File: pkgs/by-name/au/auto-editor/package.nix [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+  dependencies = [
+    numpy
+    av
+  ];
+
+  pythonImportsCheck = [ "auto_editor" ];
+
+  postInstallCheck = ''
+    $out/bin/auto-editor test all
```
Comment (dotlambda) [inline pkgs/by-name/au/auto-editor/package.nix:58]:
```suggestion
  checkPhase = ''
    runHook preCheck
  
    $out/bin/auto-editor test all
  
    runHook postCheck
```

File: pkgs/by-name/au/auto-editor/package.nix [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    # pyav is a fork of av, but has since been un-forked
+    substituteInPlace pyproject.toml \
+        --replace-fail '"pyav==12.0.5"' '"av"' \
+        --replace-fail '"ae-ffmpeg==1.2.*",' ""
+
+    # thanks to the patches above, this is no longer needed
+    rm -r ae-ffmpeg
+  '';
+
+  build-system = [ setuptools ];
```
Comment (dotlambda) [inline pkgs/by-name/au/auto-editor/package.nix:48]:
```suggestion
  build-system = with python3Packages; [ setuptools ];
```

File: pkgs/by-name/au/auto-editor/package.nix [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+  dependencies = [
```
Comment (dotlambda) [inline pkgs/by-name/au/auto-editor/package.nix:50]:
```suggestion
  dependencies = with python3Packages; [
```

File: pkgs/by-name/au/auto-editor/package.nix [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+let
+  inherit (python3Packages)
+    buildPythonApplication
+    setuptools
+    numpy
+    av
+    ;
+in
+buildPythonApplication rec {
```
Comment (dotlambda) [inline pkgs/by-name/au/auto-editor/package.nix:18]:
```suggestion
python3Packages.buildPythonApplication rec {
```

File: pkgs/by-name/au/auto-editor/package.nix [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+      src = ./set-exe-paths.patch;
+      ffmpeg = lib.getExe ffmpeg;
+      yt_dlp = lib.getExe yt-dlp;
+    })
+  ];
+
+  postPatch = ''
+    # pyav is a fork of av, but has since been un-forked
+    substituteInPlace pyproject.toml \
+        --replace-fail '"pyav' '"av'
```
Comment (dotlambda) [inline pkgs/by-name/au/auto-editor/package.nix:33]: I don't understand upstream: https://github.com/WyattBlue/auto-editor/pull/485

**Review by dotlambda — APPROVED** [review]
LGTM but I haven't tested it.

File: pkgs/by-name/au/auto-editor/package.nix [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+    $out/bin/auto-editor test all
+
+    runHook postCheck
+  '';
+
+  pythonImportsCheck = [ "auto_editor" ];
+
+  meta = {
+    description = "Command line application for automatically editing video and audio by analyzing a variety of methods, most notably audio loudness";
```
Comment (dotlambda) [inline pkgs/by-name/au/auto-editor/package.nix:57]:
```suggestion
    changelog = "https://github.com/WyattBlue/auto-editor/releases/tag${version}";
    description = "Command line application for automatically editing video and audio by analyzing a variety of methods, most notably audio loudness";
```

**Review by FliegendeWurst — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**